kernel.go

```diff
@@ -389,8 +398,8 @@ func handleExecuteRequest(ir *interp.Interp, receipt msgReceipt) error {
 	writersWG.Wait()
 
 	if executionErr == nil {
-		// if one or more value is image.Image, display it instead
-		vals = publishImages(vals, &receipt)
+		// if one or more value is image.Image or DisplayData, display it instead
+		vals = publishImagesAndDisplayData(vals, &receipt)
```

Comment:
See the [Execution Result message format](http://jupyter-client.readthedocs.io/en/stable/messaging.html#id6) which is display data plus a bit extra. This is a minor detail but if you notice the `Out[_]` is missing when a value is returned because here it is being displayed rather than put in an execution result.